vector/src/main/java/im/vector/fragments/VectorSettingsPreferencesFragment.java

```diff
@@ -1491,8 +1503,20 @@ public void onNetworkError(Exception e) {
                                                 }
 
                                                 @Override
-                                                public void onMatrixError(MatrixError e) {
-                                                    onCommonDone(e.getLocalizedMessage());
+                                                public void onMatrixError(final MatrixError e) {
+                                                    if (MatrixError.M_CONSENT_NOT_GIVEN.equals(e.errcode)) {
+                                                        if (null != getActivity()) {
+                                                            getActivity().runOnUiThread(new Runnable() {
+                                                                @Override
+                                                                public void run() {
+                                                                    ((RiotAppCompatActivity) getActivity()).getConsentNotGivenHelper().displayDialog(e);
+                                                                    hideLoadingView();
```

Comment:
I would hide the loading view before displaying the dialog

Reply:
ok